pkg/action/pull.go

```diff
@@ -109,13 +109,18 @@ func (p *Pull) Run(chartRef string) (string, error) {
 		if !filepath.IsAbs(ud) {
 			ud = filepath.Join(p.DestDir, ud)
 		}
-		if fi, err := os.Stat(ud); err != nil {
-			if err := os.MkdirAll(ud, 0755); err != nil {
+		// Let udCheck to check conflict file/dir without replacing ud when untarDir is the current directory(.).
+		udCheck := ud
+		if udCheck == "." {
+			_, udCheck = filepath.Split(chartRef)
+		}
```

Comment:
Need an else statement similar to below,  for when path is passed with `--destination` and `--untardir` flags:

```
else {
           _, chartName := filepath.Split(chartRef)
           udCheck = filepath.Join(udCheck, chartName)
}
```

Reply:
Thank you for your review. You are vary careful and I have added this change.